src/goose/toolkit/developer.py

```diff
@@ -91,6 +94,46 @@ def update_plan(self, tasks: List[dict]) -> List[dict]:
         # Return the tasks unchanged as the function's primary purpose is to update and display the task status.
         return tasks
 
+    @tool
+    def fetch_web_content(self, url: str) -> str:
+        """
+        Fetch content from a URL using available tools. Attempts to use playwright first, then falls back to wget, curl, or httpx.
+
+        Args:
+            url (str): a name of a file with the url/page content. This may be large so you can use search tools to find content. It may also contain other links which can be browsed.
+        """  # noqa
+        try:
+            with sync_playwright() as p:
+                browser = p.chromium.launch()
+                page = browser.new_page()
+                page.goto(url)
+                content = page.content()
```

Comment:
i've learned previously with web scraping projects that this will only grab static content. we usually need to add a loop to check for a DOM that signals when the relevant content is rendered (this is tricky to get right, something like a `MutationObserver`)

so not sure what we want to do here given the state of web but works great otherwise!

Reply:
forgot to leave a contrived example, my usual test website is https://www.bbc.com

`G> find all the dom elements and their class names that contain "manchester" in "https://bbc.com"`

![image](https://github.com/user-attachments/assets/dbb19018-fbbe-4e51-92a1-87c29fb566cb)